### ISSUE TITLE
SaltGUI is reporting minions v 3005.1 as offline

### ISSUE THREAD

**Issue by szmolnar** [erwindon/SaltGUI#556]
**Describe the bug**
SaltGUI is reporting minion with version 3005.1  as offline

**Expected behaviour**
Should report minion as online

**Screenshots**

<img width="491" alt="image" src="https://github.com/erwindon/SaltGUI/assets/3518047/944f2043-f32b-4bb7-b351-d696c2d550d3">
<img width="477" alt="image" src="https://github.com/erwindon/SaltGUI/assets/3518047/51448efe-a939-4f8f-ac5e-e80cfdc9ffeb">

**Comment by erwindon** [comment]
This is not an issue with saltstack version 3005.1 in general.
I have a test-bench with a VM for each individual saltstack version (but only for CentOS), and they all work fine.
Except of course for the saltstack versions that are themselves broken. For the 3xxx versions those are: 3005, 3001.5 and 3000.7.

Can you please switch to the Jobs list and select "Show all jobs".
You should be able to see the commands that SaltGUI is using internally.
for the panel "Minions" that should be `test.version` and `grains.items`. a few `wheel.` commands are also issued, but these are not registered in the jobs list.
do you see any errors in the output of {{grains.items}}?

Otherwise, are you able to inspect the developer tools from the browser? SaltGUI uses only salt-api. The REST calls for that should be visible in tab "Network". Regular salt-api calls all have the url `/`. Look ate the "Request" part of each request to see the api-call name and look at the "Response" part to see the result.

**Comment by erwindon** [comment]
As you see, PD999999 did not actually answer the `grains,items` or the `test.version` calls issued by SaltGUI.
The "Manual Run" screen does not interpret any result. So any potential mistake from the main screen cannot be present on ManualRun.
It may be that the RaspberryPi is too slow to answer the multiple requests.
Is there anything visible in the minion log from PD999999 ?

**Comment by erwindon** [comment]
can you please repeat those `salt` commands, but now using `salt --async` and started from a script, so that they run more-or-less in parallel?

and actually, the following functions are executed on the main screen (listed here in salt-command equivalent):
```
sudo salt --async "master" saltutil.wheel minions.connected
sudo salt --async "master" saltutil.wheel key.list_all
sudo salt --async "minion" grains.items
sudo salt --async "master" saltutil.wheel key.list_all
sudo salt --async "master" saltutil.runner manage.versions
sudo salt --async "master" saltutil.runner jobs.active
sudo salt --async "master" saltutil.runner jobs.list_jobs
sudo salt --async "minion" test.version
sudo salt --async "minion" saltutil.running
```
best when the script contains all these.
and then look for the results using the Job page of SaltGUI, because async commands have no output.

**Comment by erwindon** [comment]
@szmolnar ping?

**Comment by szmolnar** [comment]
Sorry for late answer. Accidentally I reconfigured my earlier system and I can’t repeat the test right now. I’m also really busy but I hope to find some time to repeat it ASAP.2024. febr. 25. dátummal, 12:57 időpontban Erwin Dondorp ***@***.***> írta:﻿
@szmolnar ping?

**Comment by erwindon** [comment]
@szmolnar thx for letting me know. I'll close this issue. You can always reopen it, or reply to it or open a new issue.
